src/main/kotlin/net/leanix/githubagent/services/CachingService.kt

```diff
@@ -8,19 +11,51 @@ import org.springframework.stereotype.Service
 class CachingService(
     private val githubEnterpriseProperties: GitHubEnterpriseProperties
 ) {
-    private val cache = HashMap<String, String?>()
 
-    @PostConstruct
-    private fun init() {
-        cache["baseUrl"] = githubEnterpriseProperties.baseUrl
-        cache["githubAppId"] = githubEnterpriseProperties.githubAppId
+    data class CacheValue(val value: Any, val expiry: Long?)
+
+    private val cache: Cache<String, CacheValue> = Caffeine.newBuilder()
+        .maximumSize(100)
+        .expireAfter(object : Expiry<String, CacheValue> {
+            override fun expireAfterCreate(
+                key: String,
+                value: CacheValue,
+                currentTime: Long
+            ): Long {
+                return value.expiry ?: Long.MAX_VALUE
+            }
+
+            override fun expireAfterUpdate(
+                key: String,
+                value: CacheValue,
+                currentTime: Long,
+                currentDuration: Long
+            ): Long {
+                return value.expiry ?: Long.MAX_VALUE
```

Comment:
Isn't the expiration value related to the token expiration date? How does this work? Do we need to take into account the token validity period? What happens if the token is expired?